crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -359,62 +359,53 @@ impl Cursor {
         false
     }
 
-    pub fn find_matching<R, F: Fn(&Cursor) -> Option<R>>(&mut self, filter_map: F) -> Option<R> {
-        while !self.is_completed {
-            if let Some(result) = filter_map(self) {
-                return Some(result);
-            }
-            self.go_to_next();
-        }
-
-        None
-    }
-
+    /// Finds the first node matching the given predicate.
+    ///
     /// In contrast to `Iterator::find_*`, this does not consume the first item when found.
-    fn find_noconsume<F: Fn(&Node) -> Option<R>, R>(&mut self, predicate: F) -> Option<R> {
+    pub fn find_matching<R, F: Fn(&Node) -> Option<R>>(&mut self, predicate: F) -> Option<R> {
         while !self.is_completed {
-            match predicate(&self.current.node) {
-                Some(result) => return Some(result),
-                _ => {
-                    self.go_to_next();
-                }
```

Comment:
I find the current version clearer, as it explicitly specifies two cases to handle:
1. If we have a `Some(result)`, return it
2. Otherwise, keep going

If we don't want to repeat `Some`, let's use an even better `match predicate(..) { result @ Some(..) => return result, .. }`

If we have to use the `if` here, I'd prefer if we used `if let Some(..)` (more obvious to me what's happening) as per above or `if result.is_some()` (simpler than matches! as it's noisier due to a macro call with a pattern vs a simple "accessor")

Reply:
Will use `is_some()` as suggested.

Reply:
I didn't know about the `result @ Some(_)` syntax!
Will use that instead. Thanks!